bpfassets/perf_event/perf_event.c

```diff
@@ -206,7 +209,7 @@ int kprobe__finish_task_switch(struct pt_regs *ctx, struct task_struct *prev)
 
     // store process metrics
     struct process_metrics_t *process_metrics;
-    process_metrics = processes.lookup(&cur_pid);
+    process_metrics = processes.lookup(&prev_tgid);
```

Comment:
per [tutorial](https://www.brendangregg.com/offcpuanalysis.html), 
```
finish_task_switch() function is called in the context of the next-running thread. 
```

and reference the oncpu analysis code [here](https://github.com/iovisor/bcc/blob/master/tools/cpudist.py)